### PR TITLE
Implement test script for US 304 - Change password function

### PR DESCRIPTION
**Pull request by Minh-Tester**
US 304-Auto_User is able to change pw
https://dev.azure.com/RookiesBatchVI/RookiesBatch6Group4/_sprints/taskboard/Rookies%20Batch%206%20-%20Group%204/RookiesBatch6Group4/Sprint%202%20-%20Finish%20manage%20asset,%20management%20assignment%20and%20manage%20request%20for%20returning%20assets?workitem=304
image

### CODE REVIEW

File: AssetManagementTestProject/TestData/FirstTimeLoginData.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        public string GetPassword()
+        {
+            string newUserDob = NewUser.Data.DateOfBirth;
+            DateTime parsedDob = DateTime.ParseExact(newUserDob, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+            string formattedDob = parsedDob.ToString("ddMMyyyy");
+            string password = GetUsername() + "@" + formattedDob;
+            return password;
+        }
+    }
+}
```
Comment (honganh1206) [inline AssetManagementTestProject/TestData/FirstTimeLoginData.cs]: Add endline here

File: AssetManagementTestProject/Services/AssetManagementAPIServices.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    (DisableUserDAO.DisableUserRequest disableUser, string token)
+    {
+        APIResponse response = DisableUserRequest(disableUser, token);
+        Assert.True(response.responseStatusCode.Equals("OK"));
+        DisableUserDAO.DisableUserResponse? message = JsonConvert.DeserializeObject
+            <DisableUserDAO.DisableUserResponse>(response.responseBody);
+        return message;
+    }
+    #endregion
+}
```
Comment (honganh1206) [inline AssetManagementTestProject/Services/AssetManagementAPIServices.cs]: Add end line

File: AssetManagementTestProject/TestCases/LoginTest.cs [hunk 3]
```diff
@@ -1,33 +1,11 @@
-﻿using AssetManagementTestProject.PageObj;
-using AssetManagementTestProject.TestSetup;
+﻿using AssetManagementTestProject.TestSetup;
 using NUnit.Framework;
 
 namespace AssetManagementTestProject;
 
 [TestFixture]
 public class LoginTest : NUnitWebTestSetup
```
Comment (honganh1206) [inline AssetManagementTestProject/TestCases/LoginTest.cs]: You don't have to do the login test, so why did you keep this file? Please remove.

File: AssetManagementTestProject/TestSetup/NUnitWebTestSetup.cs [hunk 4]
```diff
@@ -25,12 +35,40 @@ public void SetUp()
...
+        AuthorizationService = new AssetManagementAPIServices();
+        Token = AuthorizationService.ReturnLoginToken(Constant.ADMIN_USERNAME_HN, Constant.ADMIN_PASSWORD_HN);
+        APIService = new AssetManagementAPIServices();
+        NewUser = APIService.ReturnNewUser(Constant.NEW_STAFF_HN, Token);
+        FirstTimeLoginData newLoginData = new FirstTimeLoginData();
+        newLoginData.NewUser = NewUser;
+        string newAdminUsername = newLoginData.GetUsername();
+        string newAdminPassword = newLoginData.GetPassword();
+        LoginPage?.Login(newAdminUsername, newAdminPassword);
+        ChangePw1stTime = new ChangePassword1stTimePage();
```
Comment (honganh1206) [inline AssetManagementTestProject/TestSetup/NUnitWebTestSetup.cs]: Pre-Condition does not mean it should be in test setup, and what is the point of assertion in setup? I think you can include login steps into your test cases, and then add comments to let people know where your pre-condition is in your test cases

File: AssetManagementTestProject/TestData/FirstTimeLoginData.cs (hunk 1 above)
Comment (Minh-Tester) [reply AssetManagementTestProject/TestData/FirstTimeLoginData.cs]: Done

File: AssetManagementTestProject/Services/AssetManagementAPIServices.cs (hunk 2 above)
Comment (Minh-Tester) [reply AssetManagementTestProject/Services/AssetManagementAPIServices.cs]: Done

File: AssetManagementTestProject/TestCases/LoginTest.cs (hunk 3 above)
Comment (Minh-Tester) [reply AssetManagementTestProject/TestCases/LoginTest.cs]: Done

**Pull request closed without merging**